java-util/src/main/java/io/druid/java/util/common/concurrent/Execs.java

```diff
@@ -147,4 +151,30 @@ public void rejectedExecution(Runnable r, ThreadPoolExecutor executor)
         }
     );
   }
+
+  private static final LongAdder fjpWorkerThreadCount = new LongAdder();
+
+  public static ForkJoinWorkerThread makeWorkerThread(String name, ForkJoinPool pool)
+  {
+    final FJPWorkerThread t = new FJPWorkerThread(pool);
+    t.setDaemon(true);
+    fjpWorkerThreadCount.increment();
+    t.setName(StringUtils.nonStrictFormat(name, fjpWorkerThreadCount.longValue()));
+    return t;
+  }
+
+  static class FJPWorkerThread extends ForkJoinWorkerThread
```

Comment:
Early on I was considering modifying the fork join pool used here to support prioritization as per the prioritized executor service, so I had this class as a placeholder for such a scenario. Since that fell by the wayside due to lack of need for now, I'll remove this and use the standard class until such a time to where prioritization can be properly implemented and tested

Reply:
oh! also i forgot

` protected ForkJoinWorkerThread(ForkJoinPool pool)` so I have to subclass it or use one of the provided fork join pool worker types, none of which quite fit compared to just doing work in the FJP

Reply:
I'll add a comment as such

Reply:
an anonymous class works here, making the change